packages/@expo/cli/src/start/server/type-generation/index.ts

```diff
@@ -0,0 +1,38 @@
+import fs from 'fs/promises';
+import { Server } from 'metro';
+import path from 'path';
+
+import { env } from '../../../utils/env';
+import { ensureDotExpoProjectDirectoryInitialized } from '../../project/dotExpo';
+import { ServerLike } from '../BundlerDevServer';
+import { setupTypedRoutes } from './routes';
+import { forceUpdateTSConfig } from './tsconfig';
+
+export interface TypeScriptTypeGenerationOptions {
+  server: ServerLike;
+  metro: Server | null;
+  projectRoot: string;
+}
+
+export async function typescriptTypeGeneration({
+  metro,
+  projectRoot,
+  server,
+}: TypeScriptTypeGenerationOptions) {
+  if (!env.EXPO_ROUTER_TYPED_ROUTES) {
```

Comment:
```suggestion
  if (!env.EXPO_USE_TYPED_ROUTES) {
```

`EXPO_USE_` is our standard for enabling a feature, and `EXPO_NO_` is used for disabling.